### PR TITLE
Simplify training process

### PR DESCRIPTION
**Pull request by JanKapala**
Made agents training more convenient from RL scientist point of view.
Also minor import bufixing and file moving.
Closes #42

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -25,6 +25,7 @@ matplotlib = "*"
 graphviz = "*"
 parse = "*"
 bidict = "*"
+progress = "*"
```
Comment (wujuu) [inline Pipfile]: What does this lib do?

File: reinforcement_learning/common/simple_training.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        self._server = None
+        self._clients = [AgentClient(agent) for agent in self.agents]
+
+    def __enter__(self):
+        self._server = EnvironmentServer(self.engine)
+        print("Training platform has started!")
+        players = self._server.players
+        if not len(players) == len(self.agents):
+            raise AgentsNumberNotMatchPlayersNumber
+        [self._server.join(client, player) for (client, player) in zip(self._clients, players)]
```
Comment (wujuu) [inline reinforcement_learning/common/simple_training.py]: Not really in a 1970 C programming style...

File: Pipfile (hunk 1 above)
Comment (wujuu) [reply Pipfile]: I guess it shows the progress bar

File: reinforcement_learning/common/start_training.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    engine = TicTacToeEngine(2, 3, 3)
+
+    # ...and agents:
+    agents = [BasicAgent(), QLearningAgent(0.3, 0.1, 0.9)]
+
+    # You can also load previously saved agents from files:
+    agent_0_file_path = os.path.join(ABS_PROJECT_ROOT_PATH, "training_platform", "examples", "agent0.ai")
+    agent_1_file_path = os.path.join(ABS_PROJECT_ROOT_PATH, "training_platform", "examples", "agent1.ai")
+    agents_file_paths = [agent_0_file_path, agent_1_file_path]
+    agents = [BaseAgent.load(file_path) for file_path in agents_file_paths]
```
Comment (wujuu) [inline reinforcement_learning/common/start_training.py]: Big question: how does BaseAgent actually know which child class it is when loading?

File: reinforcement_learning/common/start_training.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+from reinforcement_learning.agents.basic_mc_agent.basic_mc_agent import BasicAgent
+from reinforcement_learning.agents.q_learning_agent.q_learning_agent import QLearningAgent
+from reinforcement_learning.common.simple_training import SimpleTraining
+
+if __name__ == '__main__':
+    # To start a training you need an engine:
+    engine = TicTacToeEngine(2, 3, 3)
+
+    # ...and agents:
+    agents = [BasicAgent(), QLearningAgent(0.3, 0.1, 0.9)]
```
Comment (wujuu) [inline reinforcement_learning/common/start_training.py]: Maybe we should add also optional creating of the agents here?

File: Pipfile (hunk 1 above)
Comment (JanKapala) [reply Pipfile]: Nice shot! Exactly.

File: reinforcement_learning/common/simple_training.py (hunk 2 above)
Comment (JanKapala) [reply reinforcement_learning/common/simple_training.py]: My bad :/ Doing my best...

File: reinforcement_learning/common/start_training.py (hunk 3 above)
Comment (JanKapala) [reply reinforcement_learning/common/start_training.py]: He doesn't know and he doesn't have to. The pickle lib handle it. Generaly  any object saved with pickle.dump() could be opened by BaseAgent.load which is wrapper for pickle.load().

File: reinforcement_learning/common/start_training.py (hunk 4 above)
Comment (JanKapala) [reply reinforcement_learning/common/start_training.py]: What do you mean?

File: reinforcement_learning/common/start_training.py (hunk 3 above)
Comment (wujuu) [reply reinforcement_learning/common/start_training.py]: wow

File: reinforcement_learning/common/start_training.py (hunk 4 above)
Comment (wujuu) [reply reinforcement_learning/common/start_training.py]: Instead of create_agents.py

Comment (JanKapala) [reply reinforcement_learning/common/start_training.py]: okey, I'll do

Comment (JanKapala) [reply reinforcement_learning/common/start_training.py]: Hm, after short pondering I thing that there is saving already (which is same as creating), so let's stay with this version.

Comment (wujuu) [reply reinforcement_learning/common/start_training.py]: ok

**Pull request merged**